main.tf

```diff
@@ -181,8 +181,20 @@ resource "aws_appsync_resolver" "this" {
   field  = each.value.field
   kind   = lookup(each.value, "kind", null)
 
-  request_template  = lookup(each.value, "request_template", tobool(lookup(each.value, "direct_lambda", false)) ? var.direct_lambda_request_template : "{}")
-  response_template = lookup(each.value, "response_template", tobool(lookup(each.value, "direct_lambda", false)) ? var.direct_lambda_response_template : "{}")
+  dynamic "runtime" {
```

Comment:
This can be more generically written:

```hcl
  dynamic "runtime" {
    for_each = try([each.value.runtime], [])
    
    content {
      name                   = runtime.value.name
      runtime_version = try(runtime.value.runtime_version, "1.0.0")
    }
  }
```